pkg/daemon/ceph/client/command.go

```diff
@@ -166,7 +166,7 @@ func (c *CephToolCommand) run() ([]byte, error) {
 	if command == RBDTool {
 		if c.RemoteExecution {
 			output, stderr, err = c.context.RemoteExecutor.ExecCommandInContainerWithFullOutputWithTimeout(ProxyAppLabel, CommandProxyInitContainerName, c.clusterInfo.Namespace, append([]string{command}, args...)...)
-			if stderr != "" || err != nil {
+			if err != nil {
```

Comment:
I think the `stderr` check is still important, I don't precisely remember why I did this initially but it might have been for a good reason. Especially if a Ceph CLI returns 0 but something on `stderr` it's something we have seen before. How about we treat them independently instead? So that we avoid le nil pointer when trying to access `err.Error()`.

First we check for `stderr != ""` and then `err != nil`.

Reply:
I do this, but then I start seeing this error when deploying Rook with Multus:

```
2022-01-24 17:12:35.118482 E | ceph-block-pool-controller: failed to reconcile. failed to create pool "device-health-metrics".: failed to create pool "device-health-metrics".: failed to initialize pool "device_health_metrics". : rbd: pool already registered to a different application.
```

Reply:
Should we catch and ignore `rbd: pool already registered to a different application.` then?

Reply:
I wonder if this is something that we only see with Multus given that it wasn't causing issues until then. Could there be an difference between the remote executor and local executor in terms of running this command?

Reply:
The command is the same but how we handle stdout, stderr, and err might be different and that's why we see this error. Maybe we have this error in the non-multus case but somehow it gets ignored.

Reply:
That's what I suspect as well. Let's chat in tomorrow's (Tuesday's) huddle about it more.

Reply:
The error is indeed present in the non-multus case, the CLI just returns 0:

```
[rook@rook-ceph-operator-5d4dccc85f-26bw2 /]$ rbd pool init device_health_metrics --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyringrbd: pool already registered to a different application.

[rook@rook-ceph-operator-5d4dccc85f-26bw2 /]$ echo $?
0

[rook@rook-ceph-operator-5d4dccc85f-26bw2 /]$ rbd pool init device_health_metrics --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring 2>/dev/null

[rook@rook-ceph-operator-5d4dccc85f-26bw2 /]$ rbd pool init device_health_metrics --cluster=rook-ceph --conf=/var/lib/rook/rook-ceph/rook-ceph.config --name=client.admin --keyring=/var/lib/rook/rook-ceph/client.admin.keyring 1>/dev/null
rbd: pool already registered to a different application.
```

`stderr` is still filled though, that's why the proxy (multus case) fails.